doc/api/deprecations.md

```diff
@@ -2617,6 +2617,30 @@ async function openAndClose() {
 }
 ```
 
+<a id="DEP0138"></a>
+### DEP0138: `module.parent`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: REPLACEME
+    description: Runtime deprecation.
+-->
+
+Type: Runtime
+
+A CJS module can access the first module that required it using `module.parent`.
+This feature does not work when a module is imported using ECMAScript modules
+specification, therefore it is deprecated.
+
+Some modules use it to check if they are the entry point of the current process.
+Instead, it is recommended to compare `require.main` and `module`:
+
+```js
+if (require.main === module) {
```

Comment:
Would `process.mainModule === module` be a better choice?

Reply:
yes per https://github.com/nodejs/node/blob/6de7b635a6185115975248efbc71fb8c205b59b6/doc/api/modules.md#accessing-the-main-module and https://github.com/nodejs/node/blob/6de7b635a6185115975248efbc71fb8c205b59b6/doc/api/process.md#processmainmodule usage of process.mainModule and require.main is likely preferred (though note, those values may be undefined)

Reply:
`process.mainModule === undefined` when run from an ES module, while `require.main` throws. I'd say it's probably a better idea to promote a pattern that is clearly meant for CJS. If you get `import.meta.main` for ESM, I would expect `process.mainModule` to be deprecated.

Just my opinion though, if you feel otherwise I can update the PR.